tests/aws/services/events/test_event_patterns.py

```diff
@@ -0,0 +1,160 @@
+import json
+import os
+from pathlib import Path
+from typing import List, Tuple
+
+import json5
+import pytest
+
+from localstack.testing.aws.util import is_aws_cloud
+from localstack.testing.pytest import markers
+from tests.aws.services.events.helper_functions import is_v2_provider
+
+THIS_FOLDER: str = os.path.dirname(os.path.realpath(__file__))
+REQUEST_TEMPLATE_DIR = os.path.join(THIS_FOLDER, "event_pattern_templates")
+COMPLEX_MULTI_KEY_EVENT_PATTERN = os.path.join(
+    REQUEST_TEMPLATE_DIR, "complex_multi_key_event_pattern.json"
+)
+COMPLEX_MULTI_KEY_EVENT = os.path.join(REQUEST_TEMPLATE_DIR, "complex_multi_key_event.json")
+
+
+def load_request_templates(directory_path: str) -> List[Tuple[dict, str]]:
+    json5_files = list_files_with_suffix(directory_path, ".json5")
+    return [load_request_template(file_path) for file_path in json5_files]
+
+
+def load_request_template(file_path: str) -> Tuple[dict, str]:
+    with open(file_path, "r") as df:
+        template = json5.load(df)
+    return template, Path(file_path).stem
+
+
+def list_files_with_suffix(directory_path: str, suffix: str) -> List[str]:
+    files = []
+    for root, _, filenames in os.walk(directory_path):
+        for filename in filenames:
+            if filename.endswith(suffix):
+                absolute_filepath = os.path.join(root, filename)
+                files.append(absolute_filepath)
+
+    return files
+
+
+request_template_tuples = load_request_templates(REQUEST_TEMPLATE_DIR)
+
+SKIP_LABELS = [
+    # Failing exception tests:
+    "arrays_empty_EXC",
+    "content_numeric_EXC",
+    "content_numeric_operatorcasing_EXC",
+    "content_numeric_syntax_EXC",
+    "content_wildcard_complex_EXC",
+    "int_nolist_EXC",
+    "operator_case_sensitive_EXC",
+    "string_nolist_EXC",
+    # Failing tests:
+    "complex_or",
+    "content_anything_but_ignorecase",
+    "content_anything_but_ignorecase_list",
+    "content_anything_suffix",
+    "content_exists_false",
+    "content_ignorecase",
+    "content_ignorecase_NEG",
+    "content_ip_address",
+    "content_numeric_and",
+    "content_prefix_ignorecase",
+    "content_suffix",
+    "content_suffix_ignorecase",
+    "content_wildcard_nonrepeating",
+    "content_wildcard_repeating",
+    "content_wildcard_simplified",
+    "dot_joining_event",
+    "dot_joining_pattern",
+    "nested_json_NEG",
+    "or-exists",
+    "or-exists-parent",
+]
+
+
+# TODO: extend these test cases based on the open source docs + tests: https://github.com/aws/event-ruler
+#  For example, "JSON Array Matching", "And and Or Relationship among fields with Ruler", rule validation,
+#  and exception handling.
+@pytest.mark.skipif(is_v2_provider(), reason="V2 provider does not support this feature yet")
+@pytest.mark.parametrize(
+    "request_template,label", request_template_tuples, ids=[t[1] for t in request_template_tuples]
```

Comment:
probably debatable whether it's a good idea to parameterize tests based on directory contents. It makes it very easy to add new test cases but add a small delay to test collection because it needs to list the files in the directory 🤷

Reply:
I think this is fine for tests like this :)